src/main/kotlin/org/malachite/estella/services/ApplicationService.kt

```diff
@@ -15,43 +14,42 @@ import java.util.*
 @Service
 class ApplicationService(
     @Autowired private val applicationRepository: ApplicationRepository,
-    @Autowired private val offerRepository: OfferRepository,
-    @Autowired private val jobSeekerRepository: JobSeekerRepository,
+    @Autowired private val offerService: OfferService,
+    @Autowired private val jobSeekerService: JobSeekerService,
     @Autowired private val recruitmentProcessService: RecruitmentProcessService,
     @Autowired private val interviewService: InterviewService,
     @Autowired private val mailService: MailService
-) {
-    fun insertApplicationLoggedInUser(offerId: Int, jobSeeker: JobSeeker, applicationPayload: ApplicationLoggedInPayload): Application {
-        val offer = offerRepository.findById(offerId).get()
+) : EStellaService<Application>() {
+
+    override val throwable: Exception = ApplicationNotFoundException()
+
+    fun insertApplicationLoggedInUser(
+        offerId: Int,
+        jobSeeker: JobSeeker,
+        applicationPayload: ApplicationLoggedInPayload
+    ): Application {
+        val offer = offerService.getOffer(offerId)
         val stage = offer.recruitmentProcess?.stages?.getOrNull(0)
```

Comment:
Może trochę głupie, ale czy potrzebujemy tu w ogóle tego stage'a? Raczej jak aplikujesz, to zawsze na APPLIED, czyli ten, co jest pierwszy. Raczej nie ma też możliwości, że pozwolimy HRom inaczej ustawiać, niż żeby zaczynało się APPLIED, a kończyło ENDem

Reply:
Te stage są tutaj dlatego że ważne to jest że to jest Applied dla tego konkretnego procesu a nie tylko enum Applied